### ISSUE TITLE
Android's mutation selection set is incompatible with iOS' subscription handling

### ISSUE THREAD

**Issue by eeatonaws** [aws-amplify/amplify-android#1772]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Not applicable

### Amplify Categories

DataStore

### Gradle script dependencies

N/A

### Environment information

N/A

### Describe the bug

From @Amplifiyer: 

While testing the sync between local and remote, there is some weird behavior between how it works in iOS and Android. The selection set of the create mutations in iOS seem to include the full nested model fields, while in android it’s only the ID. So for instance if I create two mutations from AppSync as follows:

```graphql
mutation OnlyIdMutation {
  createPost(input: {blogID: "6a83a3b4-2b82-4dfa-8b58-5e32326865a2", rating: 10, title: "Remote blog 6"}) {
    _deleted
    _lastChangedAt
    _version
    blogID
    created
    createdAt
    id
    rating
    title
    updatedAt
    blog {
      id
    }
  }
}
```

and

```graphql
mutation FullNestedModelMutation {
  createPost(input: {blogID: "6a83a3b4-2b82-4dfa-8b58-5e32326865a2", rating: 10, title: "Remote blog 6"}) {
    _deleted
    _lastChangedAt
    _version
    blogID
    created
    createdAt
    id
    rating
    title
    updatedAt
    blog {
      id
      updatedAt
      name
      createdAt
      _version
      _lastChangedAt
      _deleted
    }
  }
}
```

Both are successful on Android, but only the later works on iOS. The first one will result in some GraphQL errors on the iOS client:

```json
{
  "id": "2ACF986E-7607-4CD7-AEE3-148F34B01183",
  "type": "data",
  "payload": {
    "data": {
      "onCreatePost": {
        "id": "909fccc6-7ad3-4408-8b49-23616123413e",
        "created": null,
        "rating": 3,
        "title": "Post 2 android\t",
        "blog": null,
        "__typename": "Post",
        "_version": 1,
        "_deleted": null,
        "_lastChangedAt": 1606251462643
      }
    },
    "errors": [
      {
        "message": "Cannot return null for non-nullable type: 'String' within parent 'Blog' (/onCreatePost/blog/name)",
        "path": [
          "onCreatePost",
          "blog",
          "name"
        ]
      },
      {
        "message": "Cannot return null for non-nullable type: 'Int' within parent 'Blog' (/onCreatePost/blog/_version)",
        "path": [
          "onCreatePost",
          "blog",
          "_version"
        ]
      },
      {
        "message": "Cannot return null for non-nullable type: 'AWSTimestamp' within parent 'Blog' (/onCreatePost/blog/_lastChangedAt)",
        "path": [
          "onCreatePost",
          "blog",
          "_lastChangedAt"
        ]
      }
    ]
  }
}
```

### Additional information and screenshots

A few notes on the issue:

- We have a [toggle for this behavior](https://github.com/aws-amplify/amplify-android/blob/main/aws-api-appsync/src/main/java/com/amplifyframework/api/aws/LeafSerializationBehavior.java#L21), when making GraphQL requests. The DataStore category [only includes requests IDs from children](https://github.com/aws-amplify/amplify-android/blob/main/aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/DataStoreGraphQLRequestOptions.java#L63), and the [API category requests all fields](https://github.com/aws-amplify/amplify-android/blob/main/aws-api/src/main/java/com/amplifyframework/api/aws/ApiGraphQLRequestOptions.java#L56).
- It’s breaking in iOS most likely because the underlying model is required or has required properties. Swift is very strict when deserializing JSON since types declared as required must be provided at creation time. You can’t do `name: String!` and then not provide it when constructing the object.

**Comment by MorCohenAres** [comment]
Having the same issue.
This actually breaks iOS DataStore subscriptions completely... Any plans on adding the `LeafSerializationBehavior` toggle to iOS ?

**Comment by chrisbonifacio** [comment]
Hi @MorCohenAres 👋 the specification work to address this issue has been completed, it is planned to be released in the near future.

**Comment by AnilMaktala** [comment]
@PS-MS @MorCohenAres we are currently on a feature that will resolve this issue. we will update the ticket once it is released.

**Comment by jlindsay-ps** [comment]
Is there any update to this issue? It's been more than 6 months since the last update. We're still having to query each table individually because we are unable to use connections.

**Comment by eeatonaws** [comment]
A [fix](https://github.com/aws-amplify/amplify-swift/pull/2583) was released in Amplify Swift 2.4.0 to modify the selection set on iOS, making the Android and iOS selection sets compatible. Please update to Amplify Swift 2.4.0 or newer and follow the [Amplify Swift selection set documentation](https://docs.amplify.aws/cli/migration/lazy-load-custom-selection-set/) for additional information.

**Comment by eeatonaws** [comment]
Closing since the issue has been resolved. Please let us know if you have any additional questions.
